pkg/driver/azure-disk/azure_disk.go

```diff
@@ -0,0 +1,411 @@
+
+	dc "github.com/openshift/library-go/pkg/operator/deploymentcontroller"
+
+	opCfgV1 "github.com/openshift/api/config/v1"
+	opv1 "github.com/openshift/api/operator/v1"
+	commongenerator "github.com/openshift/csi-operator/pkg/driver/common/generator"
+	"github.com/openshift/library-go/pkg/operator/configobserver/featuregates"
+	appsV1 "k8s.io/api/apps/v1"
+	coreV1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+)
+
+const (
+	cloudCredSecretName   = "azure-disk-credentials"
+	metricsCertSecretName = "azure-disk-csi-driver-controller-metrics-serving-cert"
+	infrastructureName    = "cluster"
+	cloudConfigName       = "kube-cloud-config"
+	caBundleKey           = "ca-bundle.pem"
+	trustedCAConfigMap    = "azure-disk-csi-driver-trusted-ca-bundle"
+
+	configMapName                        = "cloud-provider-config"
+	openshiftDefaultCloudConfigNamespace = "openshift-config"
+
+	generatedAssetBase = "overlays/azure-disk/generated"
+
+	operatorImageVersionEnvVarName = "OPERATOR_IMAGE_VERSION"
+	ccmOperatorImageEnvName        = "CLUSTER_CLOUD_CONTROLLER_MANAGER_OPERATOR_IMAGE"
+
+	// for azure stack hub
+	configEnvName         = "AZURE_ENVIRONMENT_FILEPATH"
+	azureStackCloudConfig = "/etc/azure/azurestackcloud.json"
+	azureCfgName          = "azure-cfg"
+
+	diskEncryptionSetID = "diskEncryptionSetID"
+
+	// for azure hypershift install
+	hypershiftCloudConfigName = "azure-cloud-config"
+)
+
+func GetAzureDiskGeneratorConfig() *generator.CSIDriverGeneratorConfig {
+	return &generator.CSIDriverGeneratorConfig{
+		AssetPrefix:      "azure-disk-csi-driver",
+		AssetShortPrefix: "azure-disk",
+		DriverName:       "disk.csi.azure.com",
+		OutputDir:        generatedAssetBase,
+
+		ControllerConfig: &generator.ControlPlaneConfig{
+			DeploymentTemplateAssetName: "overlays/azure-disk/patches/controller_add_driver.yaml",
+			LivenessProbePort:           10301,
+			MetricsPorts: []generator.MetricsPort{
+				{
+					LocalPort:           commongenerator.AzureDiskLoopbackMetricsPortStart,
+					InjectKubeRBACProxy: true,
+					ExposedPort:         commongenerator.AzureDiskExposedMetricsPortStart,
+					Name:                "driver-m",
+				},
+			},
+			SidecarLocalMetricsPortStart:   commongenerator.AzureDiskLoopbackMetricsPortStart + 1,
+			SidecarExposedMetricsPortStart: commongenerator.AzureDiskExposedMetricsPortStart + 1,
+			Sidecars: []generator.SidecarConfig{
+				commongenerator.DefaultProvisionerWithSnapshots.WithExtraArguments(
+					"--default-fstype=ext4",
+					"--feature-gates=Topology=true",
+					"--extra-create-metadata=true",
+					"--timeout=60s",
+				),
+				commongenerator.DefaultAttacher.WithExtraArguments(
+					"--timeout=60s",
+				),
+				commongenerator.DefaultResizer.WithExtraArguments(
+					"--timeout=300s",
+				),
+				commongenerator.DefaultSnapshotter.WithExtraArguments(
+					"--timeout=300s",
```

Comment:
This is probably an oversight, but the original operator did not set any timeout.
Upstream has `            - "--timeout=1200s"`